src/target/arc_ocd.c

```diff
@@ -141,6 +141,13 @@ int arc_ocd_deassert_reset(struct target *target)
 	/* deassert reset lines */
 	jtag_add_reset(0, 0);
 
+	if (!target->reset_halt && target->state == TARGET_RESET)
+	{
+		CHECK_RETVAL(target_halt(target));
```

Comment:
Why do you need this? To safely set breakpoints in subsequent **target_resume()** while processor is not running? Pls add a comment. You might merge this comment with the one to **target_resume()** call, if you wish.